pkg/kudoctl/cmd/init.go

```diff
@@ -110,23 +116,30 @@ func (initCmd *initCmd) run() error {
 	//TODO: implement output=yaml|json (define a type for output to constrain)
 	//define an Encoder to replace YAMLWriter
 	if strings.ToLower(initCmd.output) == "yaml" {
-		mans, err := cmdInit.PrereqManifests(opts)
-		if err != nil {
-			return err
+
+		var mans []string
+		if !initCmd.crdOnly {
+			prereq, err := cmdInit.PrereqManifests(opts)
+			if err != nil {
+				return err
+			}
+			mans = prereq
 		}
 
 		crd, err := cmdInit.CRDManifests()
 		if err != nil {
 			return err
 		}
+		mans = append(mans, crd...)
```

Comment:
I think at this point, mans is uninitialized slice so it will fail when using crdonly.

Reply:
append works on the zero value of a slice (which is nil). Great GopherCon talk that dives into making the zero value useful: https://www.youtube.com/watch?v=ynoY2xz-F8s

https://play.golang.org/p/fli3pv-4D3v